onnxruntime/test/providers/cpu/tensor/gather_op_test.cc

```diff
@@ -101,7 +101,7 @@ TEST(GatherOpTest, Gather_invalid_index_cpu) {
   ASSERT_STATUS_OK(so.config_options.AddConfigEntry(kOrtSessionOptionsConfigStrictShapeTypeInference, "0"));
   test.Run(so, OpTester::ExpectResult::kExpectFailure, "indices element out of data bounds, idx=1000 must be within the inclusive range [-3,2]",
            // On Cuda it is impossible to dereference indices memory on CPU so the check can not run
-           {kCudaExecutionProvider, kOpenVINOExecutionProvider, kDnnlExecutionProvider, kNupharExecutionProvider, kTensorrtExecutionProvider});
+           {kCudaExecutionProvider, kOpenVINOExecutionProvider, kDnnlExecutionProvider, kNupharExecutionProvider, kTensorrtExecutionProvider, kNnapiExecutionProvider});
```

Comment:
nit: may be better to use the Run argument that specifies which EPs to use, instead of those to exclude
https://github.com/microsoft/onnxruntime/blob/972e5e73008a8af20eab47905d5acbbf00d1373b/onnxruntime/test/providers/provider_test_utils.h#L733

Reply:
In general it's better to exclude so that new EPs get tested by default.

Reply:
yeah, generally I agree. I think this test is targeting the CPU EP as it's checking for a specific error message and it has "cpu" in the name.